build.gradle

```diff
@@ -60,6 +60,23 @@ tasks.named("test") {
     }
     exceptionFormat "full"
   }
+  // AGP 8+ requires JVM 17+.
+  javaLauncher = javaToolchains.launcherFor {
+    languageVersion = JavaLanguageVersion.of(17)
```

Comment:
Use 17 on CI too?

Reply:
Could you please elaborate why?

Reply:
Gradle will reuse the installed JDK first, https://docs.gradle.org/current/userguide/toolchains.html#sec:auto_detection

Reply:
The used JDK to execute Gradle is independent of the JDK used to run the compiler or tests.
To actually run Gradle (this project), you need a JVM, depending on your setup either JAVA_HOME, your project JDK (IntelliJ) or the the specified JDK in GitHub workflow (using setup-java setting JAVA_HOME).
The used Gradle version + Gradle plugins by licensee requires JVM 11.
Now licensee specified JVM 11 for compiling, without any constraints. The same version is used on CI, so Gradle reuse the JDK.
The tests requires JDK 17 (due to AGP 8+), which is specified in the test task using the javaLauncher. So Gradle needs get a JDK 17.
The auto-detection you mention is about the default paths Gradle looks for JDK 17. The hostedtoolchain paths on GitHub CI isn't used today by Gradle (https://github.com/gradle/gradle/issues/14903), so we use the foojay resolver plugin. Without this plugin, Gradle does not find installed JDKs.

Reply:
TIL, I always believe Gradle honored the output path of `setup-java`, thanks for your sharing.

Reply:
Unfortunately, it doesn't, but this will finally be fixed by the grade-build-action, according to the planned 2.8.0 release: https://github.com/gradle/gradle-build-action/issues/561

Reply:
Yep, good to know.